src/test/java/com/google/api/generator/gapic/composer/common/TestProtoLoader.java

```diff
@@ -218,6 +219,31 @@ public GapicContext parseShowcaseTesting() {
         .build();
   }
 
+  public GapicContext parseExplicitDynamicRoutingHeaderTesting() {
+    FileDescriptor testingFileDescriptor = ExplicitDynamicRoutingHeaderTestingOuterClass.getDescriptor();
+    ServiceDescriptor testingService = testingFileDescriptor.getServices().get(0);
+    assertEquals(testingService.getName(), "ExplicitDynamicRoutingHeaderTesting");
+
+    Map<String, Message> messageTypes = Parser.parseMessages(testingFileDescriptor);
+    Map<String, ResourceName> resourceNames = Parser.parseResourceNames(testingFileDescriptor);
+    Set<ResourceName> outputResourceNames = new HashSet<>();
```

Comment:
Just to confirm the order doesn't matter. If it matters, consider using `LinkedHashSet`.

Reply:
No, it does not matter